handler/producer/sequence.go

```diff
@@ -52,12 +70,28 @@ func pipe(req *http.Request, rt []Roundtrip, kind string) chan *Result {
 	for _, srt := range rt {
 		rch := make(chan *Result, srt.Len())
 		allResults = append(allResults, rch)
+		k := fmt.Sprintf("%v", srt.Names())
+		if val, ok := additionalChs.Load(k); ok {
```

Comment:
I think this section is hard to read and unterstand. `rch` was initially the *main* result channel but counting them up and even using the counted one as named argument in `pipeResults` is not helpful. So let's find a better naming for this.
Also the inner loop scope has similar code as the outer one. Create same map, same switch case if there is no other way let's use functions for this.